### PR TITLE
Allow `kedro new` with config file without tools or example

### LINKED ISSUE #3342: Make it possible to create project from config without providing `add_ons` or `example_pipeline`
## Description
Creating a project from config without providing `add_ons` or `example_pipeline` fails.

## Context
`config.yml`:

```yaml
project_name: My First Kedro Project
repo_name: testing-kedro
python_package: test_kedro
```
By itself, this doesn't work - throws the following error:
```
kedro.framework.cli.utils.KedroCliError: example_pipeline, add_ons not found in config file.
Run with --verbose to see the full exception
```
I would assume these would use the default values if not provided.

## Expected behaviour
The above config yaml is valid and so project creation should succeed.

### PR DESCRIPTION
**Pull request by ankatiyar**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Fix #3342 

## Development notes

- Only fail if `project_name` is not in `config.yml`
- Display a message saying default value is being used for "missing keys" i.e. either tools or example_pipeline
- Added a test

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/framework/cli/starters.py [hunk 1]
```diff
@@ -704,9 +704,14 @@ def _validate_config_file_against_prompts(
     if config is None:
         raise KedroCliError("Config file is empty.")
     missing_keys = set(prompts) - set(config)
-    if missing_keys:
+    if "project_name" in missing_keys:
```
Comment (DimedS) [inline kedro/framework/cli/starters.py:707]: thanks @ankatiyar , great job!
Maybe we should make a slight change, like this:
missing_keys_without_mandratory = missing_keys - set(["example_pipeline", "tools"])
if missing_keys_without_mandratory:
        click.echo(yaml.dump(config, default_flow_style=False))
        raise KedroCliError(f"{', '.join(missing_keys)} not found in config file.")
This change will allow us to check not only the project_name but also all other prompts that users are waiting for (except for the mandatory example_pipeline and tools), similar to how it was previously.

File: kedro/framework/cli/starters.py [hunk 2]
```diff
@@ -704,9 +704,14 @@ def _validate_config_file_against_prompts(
...
         click.echo(yaml.dump(config, default_flow_style=False))
         raise KedroCliError(f"{', '.join(missing_keys)} not found in config file.")
+    if "example_pipeline" in missing_keys:
```
Comment (DimedS) [inline kedro/framework/cli/starters.py:710]: Why are we checking only for `example_pipeline` and not `tools`? I think if `tools` will not be provided we also will use default values?

Comment (ankatiyar) [reply kedro/framework/cli/starters.py:710]: Changing this!

File: kedro/framework/cli/starters.py (hunk 1 above)
Comment (ankatiyar) [reply kedro/framework/cli/starters.py]: Updated!

**Review by SajidAlamQB — APPROVED** [review]
Thank you @ankatiyar! 👍

**Pull request merged**